geospaas_harvesting/verify.py

```diff
@@ -0,0 +1,51 @@
+""" Verification module. """
+import os
+from datetime import datetime
+
+import django
+import requests
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_harvesting.settings')
+django.setup()
+
+from geospaas.catalog.models import DatasetURI
+
+def main():
+    """ Verifies the datasets based on their dataseturi. If the download link does not provide a
+    download availability and returns a text or html response, then the dataset uri is removed. If
+    there is no other "dataseturi" remains for the dataset, then the dataset is also removed in
+    order to be harvested again in the future with a correct and healthy "dataseturi".
+    Since the number of datasets in the database might be enormous, the datasets are retrieved
+    into with a variable named retrieved_dataset_uris with an specific length number
+    for memory management. """
+    corrupted_url_set = set()
+    id_range = range(DatasetURI.objects.earliest('id').id,
+                     DatasetURI.objects.latest('id').id, 1000)# <=number for the length of retrieved
+    for i in range(len(id_range)):
+        try:
+            retrieved_dataset_uris = DatasetURI.objects.filter(
+                id__gte=id_range[i], id__lt=id_range[i+1])
+        except IndexError:
+            retrieved_dataset_uris = DatasetURI.objects.filter(
+                id__gte=id_range[i], id__lte=DatasetURI.objects.latest('id').id)
+        for dsuri in retrieved_dataset_uris:
+            content_type = requests.head(dsuri.uri, allow_redirects=True).headers.get('content-type')
```

Comment:
Any particular reason to get the Content-Type header (which might not be set) instead of relying on the status code?

Reply:
@aperrin66 
How we are going to identify a download link from its status code? What would that status be for a download link? is it distinguishable from a `text` or `html` response that might describe the failure of the download link instead of download link itself?

The reason for using content type is that it clearly shows that this link is ready for starting a download action. It shows it by setting the content type value as, for example,`application/x-netcdf;charset=ISO-8859-1`

Reply:
If the status code is different from 2**, there is a problem.